packages/application-components/src/components/modal-page/modal-page.js

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Modal from 'react-modal';
+import { PORTALS_CONTAINER_ID } from '@commercetools-frontend/constants';
+import {
+  Spacings,
+  Text,
+  Card,
+  SecondaryIconButton,
+  CloseIcon,
+} from '@commercetools-frontend/ui-kit';
+import styles from './modal-page.mod.css';
+
+// When running tests, we don't render the AppShell. Instead we mock the
+// application context to make the data available to the application under
+// test. However, this also means that the AppShell tree is not rendered,
+// which contains among other things the <div id="portals-container" />
+// element, used to render the modal content.
+// Apparently though, when the <Modal> component unmounts between tests, it
+// tries to remove the child elements within the parent selector
+// ("portals-container") which may have been cleaned up by another test,
+// resulting in the <Modal> to throw an NotFoundError.
+// To solve this, we can simply use "document.body" as the parent selector
+// instead of a specific element that will be cleaned up, resulting in
+// console errors (even though the test passes). We only need to to this in
+// test environment.
+const parentSelector = () =>
+  process.env.NODE_ENV === 'test'
+    ? document.body
+    : document.querySelector(`#${PORTALS_CONTAINER_ID}`);
+
+const getZIndexForLevel = (baseZIndex, level) => {
+  switch (level) {
+    case 'two':
+      return baseZIndex + 1;
+    case 'three':
+      return baseZIndex + 2;
+    default:
+      return baseZIndex;
+  }
+};
+
+const getColumnSpacing = level => {
+  const baseSpacing = 64;
+  switch (level) {
+    case 'two':
+      return `${baseSpacing}px ${baseSpacing}px auto`;
+    case 'three':
+      return `${baseSpacing}px ${baseSpacing}px ${baseSpacing}px auto`;
+    default:
+      return `${baseSpacing}px auto`;
+  }
+};
+
+const getStyles = props => ({
+  overlay: {
+    zIndex: props.zIndex || getZIndexForLevel(props.baseZIndex, props.level),
+  },
+  content: {
+    gridTemplateColumns: getColumnSpacing(props.level),
+  },
+});
+
+const Spacers = props => {
+  switch (props.level) {
+    case 'two':
+      return (
+        <React.Fragment>
+          <div />
+          <div />
+        </React.Fragment>
+      );
+    case 'three':
+      return (
+        <React.Fragment>
+          <div />
+          <div />
+          <div />
+        </React.Fragment>
```

Comment:
Can you elaborate what happens here and why we render those empty divs?

Reply:
These are for adding more space to the left of the component, with a grey semi-transparent overlay. The space should programmatically increase with the level of the modal.

However, the way this is done will change, as there will be more than 3 levels, and we'll rethink the logic.